app/js/editQuestion.js

```diff
@@ -0,0 +1,98 @@
+    return await sendData(jsonToFormData(newQuestionData), `/question/${id}/edit`);
+}
+
+async function getQuestionById(id) {
+    const { data } = await getData("/question");
+    return data.filter(question => question.id === String(id));
+}
+
+async function getAnswerByQuestionId(id) {
+    const { data } = await getData(`/answer/${id}`);
+    return data.answer;
+}
+
+// We are listening to the load event so that we can use async/await.
+window.addEventListener("load", async () => {
+    const form = document.querySelector("#edit-question");
+    const questionId = location.hash.replace("#", "");
+    const [question] = await getQuestionById(questionId);
+    const answer = await getAnswerByQuestionId(questionId);
+
+    document.querySelector("#question-to-edit").value = question.text;
+    document.querySelector("#option1").value = question.option1;
+    document.querySelector("#option2").value = question.option2;
+    document.querySelector("#option3").value = question.option3;
+    document.querySelector("#option4").value = question.option4;
+    document.querySelector("#option5").value = question.option5;
+    document.querySelector("#test_id").value = question.test_id;
+    document.querySelector(`input[data-question-id="${answer}"]`).checked = 1;
+
+    form.addEventListener("submit", async (event) => {
+       event.preventDefault();
+
+       const responseMsg = document.querySelector('#inputSubmissionConfirmation');
+       const hasQuestions = formHasQuestion(form);
+       const hasAnswers = formHasBetweenTwoAndFiveAnswers(form);
+       const isValid = answerHasValidValue(form);
+
+        if (hasQuestions && hasAnswers && isValid) {
+            let answer;
+
+            if (!authorised) {
+                return;
+            }
+
+            document.querySelectorAll('.answer-check').forEach((answerCheck, index) => {
+                if (answerCheck.checked) {
+                    answer = index + 1;
+                }
+            });
+
+            const { success, message } = await editQuestion(questionId,{
+                text: document.querySelector("#question-to-edit").value,
+                option1: document.querySelector("#option1").value,
+                option2: document.querySelector("#option2").value,
+                option3: document.querySelector("#option3").value,
+                option4: document.querySelector("#option4").value,
+                option5: document.querySelector("#option5").value,
+                test_id: document.querySelector("#test_id").value,
+                answer
+            });
+
+            responseMsg.textContent = message;
+
+            if (success){
+                responseMsg.classList.add('alert-success');
+                responseMsg.classList.remove('alert-danger');
+            } else {
+                responseMsg.classList.remove('alert-success');
+                responseMsg.classList.add('alert-danger');
+            }
+        } else {
```

Comment:
I'm probably wrong, but does this last else statement need bumping to the left by one?